backend-rust/src/indexer/ensure_affected_rows.rs

```diff
@@ -0,0 +1,78 @@
+//! This module contains the extension trait `EnsureAffectedRows` adding
+//! validation methods on the query result.
+
+use std::ops::RangeInclusive;
+
+/// Extension trait providing methods for validating number of affected rows of
+/// a query result.
+/// Only implemented for `sqlx::postgres::PgQueryResult`.
+pub trait EnsureAffectedRows: Sized {
+    /// Validate number of affected rows are exactly as expected for a query
+    /// result.
+    fn ensure_affected_rows(
+        self,
+        expected_count: u64,
+    ) -> Result<Self, EnsureAffectedRowsExactError>;
+    /// Validate number of affected rows are within an expected range for a
+    /// query result.
+    fn ensure_affected_rows_in_range(
+        self,
+        expected_range: RangeInclusive<u64>,
+    ) -> Result<Self, EnsureAffectedRowsInRangeError>;
+
+    /// Validate number of affected rows are exactly 1 for a query result.
+    fn ensure_affected_one_row(self) -> Result<Self, EnsureAffectedRowsExactError> {
+        self.ensure_affected_rows(1)
+    }
+}
+
+impl EnsureAffectedRows for sqlx::postgres::PgQueryResult {
+    fn ensure_affected_rows(self, expected: u64) -> Result<Self, EnsureAffectedRowsExactError> {
+        let affected = self.rows_affected();
```

Comment:
You could just call ensure_affected_rows_in_range here

Reply:
I considered that as well, but the error message becomes less nice